Lib/test/test_abstract_numbers.py

```diff
@@ -40,5 +60,136 @@ def test_complex(self):
         self.assertRaises(TypeError, int, c1)
 
 
+class TestNumbersDefaultMethods(unittest.TestCase):
+    def test_complex(self):
+        @concretize
+        class MyComplex(Complex):
+            def __init__(self, real, imag):
+                self.r = real
+                self.i = imag
+
+            @property
+            def real(self):
+                return self.r
+
+            @property
+            def imag(self):
+                return self.i
+
+            def __add__(self, other):
+                if isinstance(other, Complex):
+                    return MyComplex(self.imag + other.imag,
+                                     self.real + other.real)
+                if isinstance(other, Number):
+                    return MyComplex(self.imag + 0, self.real + other.real)
```

Comment:
Could not it be simplified?
```suggestion
                if isinstance(other, Number):
                    return MyComplex(self.imag + other.imag,
                                     self.real + other.real)
                raise NotImplementedError
```

Reply:
On other hand, `real` and `imag` are only defined for `Complex`, so it should be
```suggestion
                if isinstance(other, Complex):
                    return MyComplex(self.imag + other.imag,
                                     self.real + other.real)
                raise NotImplementedError
```